src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolManagerBase.cs

```diff
@@ -337,16 +331,18 @@ public bool Equals(TypeArrayKey other)
                 if (other._types == _types) return true;
                 if (other._types.Length != _types.Length) return false;
                 if (other._hashCode != _hashCode) return false;
-                for (int i = 0, n = _types.Length; i < n; i++)
+                for (int i = 0; i < _types.Length; i++)
                 {
                     if (!_types[i].Equals(other._types[i]))
                         return false;
                 }
                 return true;
             }
 
+            [ExcludeFromCodeCoverage] // Typed overload should always be the method called.
```

Comment:
But it is a gap: it's code in the product that's not being tested.  That could mean that's because it's not reachable or that we're missing tests, but this seems to be a middle-ground: we've not done the due diligence to be definitively sure it'll never be used, so we still have the code in the body, but then attributing it in a way that'll cause us to ignore it.